### PR TITLE
fix: save dtypes, index-dtype, and columns-dtype in ModelJSONSerializer

### PR DESCRIPTION
**Pull request by tcrasset**
- fix: save dtypes, index-dtype, and columns-dtype in ModelJSONSerializer
- chore: misc changes

Closes https://github.com/octopize/avatar/issues/887

### CODE REVIEW

File: saiph/serializer.py [hunk 1]
```diff
@@ -38,12 +48,51 @@ def default(self, obj):
...
             data = obj.to_json(orient="index", default_handler=str)
-            return dict(__frame__=data)
+
+            # We cast to string and not to object, to transform the
+            # 'inferred_type' property (read-only) to string
+            # If we don't it remains at integer when decoding and using
+            # an index or column with `'0'`
+            index_type = (
+                "str" if str(obj.index.dtype) == "object" else str(obj.index.dtype)
+            )
```
Comment (tas17) [inline saiph/serializer.py:84]: Aren't you doing this twice ?

**Review by tas17 — COMMENTED** [review]
Found it pretty hard to follow, but I think it's usual on this kind of changes...

File: saiph/serializer.py (hunk 1 above)
Comment (tcrasset) [reply saiph/serializer.py:84]: Yes, but I need to be inside the if-statements so that other types of objects (e.g. str) do not execute that line

Comment (tcrasset) [reply saiph/serializer.py:84]: Yes, but I need to be inside the if-statements so that other types of objects (e.g. str) do not execute that line

**Review by jpetot — APPROVED** [review]
tested and it works :)
